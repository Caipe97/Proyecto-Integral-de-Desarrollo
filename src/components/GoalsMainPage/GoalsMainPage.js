import React, { Component } from 'react';
import '../../pages/Goals/Goals.css';
import Header from '../Header/Header';
import Footer from '../Footer/Footer';
import FoodCategoriesSearchBar from '../SearchBar/FoodCategoriesSearchBar';
import DatePicker from "react-datepicker";
import TextField from '@material-ui/core/TextField';
import Goal from '../../images/goal.png';
import Delete from '../../images/delete.png'

class GoalsMainPage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      name: '',
      totalCalories: '',
      dateStart: new Date(),
      dateStartOrigin: new Date(),
      successMessage: ''
    };
  }

  async componentDidMount(){
    if(this.props.userId === 0){
      await this.props.onRefreshPage();
      this.props.history.push("/profile");
    }
    if(this.props.history.location.state){
      let copyDateStart = new Date(this.props.history.location.state.goal.dateStart)
      let sumThreeHoursToDateStart = copyDateStart.setHours(copyDateStart.getHours() + 3)
      await this.setState({
        ...this.state,
        name: this.props.history.location.state.goal.name,
        totalCalories: this.props.history.location.state.goal.totalCalories,
        dateStart: sumThreeHoursToDateStart,
        dateStartOrigin: new Date(sumThreeHoursToDateStart)
      })
    } else {
      this.props.onResetCurrentGoal();
    }  
  }

  validateQuantity = (quantity) => {//cannot be empty and only numbers
    const expression = /^\d+$/
    return (quantity === '' || expression.test(String(quantity).toLowerCase()))
  }

  handleChangeName = async event => {
    await this.setState({
      ...this.state,
      [event.target.name]: event.target.value,
      successMessage: ''
    }) 
  }

  handleChangeCalories = async event => {
    if(this.validateQuantity(event.target.value)){
      await this.setState({
        ...this.state,
        [event.target.name]: event.target.value,
        successMessage: ''
      })
    } else {
      await this.setState({quantity: ''})
    }
  };

  changeDateStartFormat = date => {
    date = new Date(date)
    if(date){
      const year = date.getFullYear();
      const month = date.getMonth() + 1;
      if(month < 10){
        return (year+'-0'+month);
      } else {
        return (year+'-'+month);
      }
    }
    return '';
  }

  objectivesSumedCaloriesIsSmallerOrEqualThanGoalTotalCalories = (goal, totalCalories) => {
    if(this.state.totalCalories !== 0){
      const objectivesTotalCalories = goal.objectives.reduce((totalCalories, objective) => {
        return totalCalories + objective.objectiveCalories
      }, 0)
      if(objectivesTotalCalories > totalCalories){
        return false;
      } else{
        return true;
      }
    } else {
      if(goal.objectives.length === 0){
        return false;
      }
      return true;
    }
  }

  handleSubmitCreate = async event => {
    event.preventDefault();
    if(this.state.totalCalories === ''){
      await this.setState({
        ...this.state,
        totalCalories: 0
      })
    }
    if(this.objectivesSumedCaloriesIsSmallerOrEqualThanGoalTotalCalories(this.props.currentGoal, this.state.totalCalories)){
      if(this.isValidDate(this.state.dateStart, false)){
        if(this.state.name !== ''){
          let neccesaryDateStartFormat = this.changeDateStartFormat(this.state.dateStart);
          this.props.onChangeCurrentGoalNameTotalCaloriesAndDateStart(this.state.name, this.state.totalCalories, neccesaryDateStartFormat);
          const data = await this.props.onAddGoal(this.props.userId, this.props.currentGoal);
          if(data.payload.message){
            this.setState({
              ...this.state,
              successMessage: 'Completa todos los campos para cargar una meta'
            })
          } else {
            this.setState({
              ...this.state,
              successMessage: 'Meta agregada exitosamente'
            })
          }
          this.setState({name: '', totalCalories: '', dateStart: '', dateStartOrigin: ''});
          this.props.onResetCurrentGoal();
        } else {
          this.setState({
            ...this.state,
            successMessage: 'La meta debe tener un nombre'
          })
        }
      } else {
        this.setState({
          ...this.state,
          successMessage: 'Fecha invalida'
        })
      }
    } else{
      this.setState({
        ...this.state,
        successMessage: 'La suma de las calorias de los objetivos no puede superar a las calorias totales de la meta'
      })
    }
  };

  isValidDate = (date, isEdit) => {
    let isValidDate = true;
    let dateCopy = date.getMonth() + '' + date.getFullYear();
    this.props.goals.forEach(goal => {
      let goalDateStartCopy = new Date(new Date(goal.dateStart).setHours(new Date(goal.dateStart).getHours() + 3))
      goalDateStartCopy = goalDateStartCopy.getMonth() + '' + goalDateStartCopy.getFullYear();
      if(goalDateStartCopy === dateCopy){
        let dateStarCopy = new Date(this.state.dateStartOrigin);
        dateStarCopy = dateStarCopy.getMonth() + '' + dateStarCopy.getFullYear()
        if(isEdit){
          if(goalDateStartCopy !== dateStarCopy){
            isValidDate = false;
            return isValidDate;
          }
        } else {
          isValidDate = false;
          return isValidDate;
        }
      } 
    })
    return isValidDate
  }

  handleSubmitUpdate = async event => {
    event.preventDefault();
    if(this.state.totalCalories === ''){
      await this.setState({
        ...this.state,
        totalCalories: 0
      })
    }
    if(this.objectivesSumedCaloriesIsSmallerOrEqualThanGoalTotalCalories(this.props.currentGoal, this.state.totalCalories)){
      if(this.isValidDate(new Date(this.state.dateStart), true)){
        let neccesaryDateStartFormat = this.changeDateStartFormat(this.state.dateStart);
        this.props.onChangeCurrentGoalNameTotalCaloriesAndDateStart(this.state.name, this.state.totalCalories, neccesaryDateStartFormat);
        const data = await this.props.onUpdateCurrentGoal(this.props.userId, this.props.currentGoal);
        if(data.payload.message){
          this.setState({
            ...this.state,
            successMessage: 'Completa todos los campos para editar una meta'
          })
        } else {
          this.setState({
            ...this.state,
            successMessage: 'Meta editada exitosamente'
          })
        }
        this.setState({name: '', totalCalories: '', dateStart: '', dateStartOrigin: ''});
        this.props.onResetCurrentGoal();
      } else {
        this.setState({
          ...this.state,
          successMessage: 'Fecha invalida'
        })
      }
    } else{
      this.setState({
        ...this.state,
        successMessage: 'La suma de las calorias de los objetivos no puede superar a las colarias totales de la meta'
      })
    }
  };

  handleChangeDateStart = date => {
    const today = new Date();
    if(date < today){
      date = today;
    }
    this.setState({
      ...this.state,
      dateStart: date
    })
  };
  
  getFoodCategoryNameOfObjective = (objective) => {
    let categoryName;
    this.props.foodCategories.forEach(foodCategory => {
      if(objective.foodCategoryId === foodCategory.foodCategoryId){
        categoryName = foodCategory.name;
      }
    });
    return categoryName;
  }

  render() {
    return (
      <div className="contenedorR">
        <Header {...this.props}/>
        <div className='contenidoR' style={{backgroundColor:'#B6E052'}}>
          <div className="col2R">
            <FoodCategoriesSearchBar {...this.props}/>
          </div>
        </div>
        <div className="sidebarR" >
          <div style={{ display: 'flex', justifyContent: 'center', alignItems: 'center', flexDirection: 'column' }}>
            <img src={Goal} alt="goal" style={{width: 100}}/>
            <div style={{ marginTop: 15 }}>
            <TextField label="Nombre de la meta" name='name' type='name' value={this.state.name} onChange={this.handleChangeName} required/>
            <TextField label="CalorÃ­as de la meta" name='totalCalories' type='totalCalories' value={this.state.totalCalories} onChange={this.handleChangeCalories} required/>
            </div>
            <div className="row">
              <div className="w-100"></div>
              <div className="col" style={{ textAlign: 'center' }}>
                <ul style={{ marginBlock: '0em', paddingInlineStart: '1%' }}>
                  {this.props.currentGoal.objectives.map((objective) =>
                    <div key={objective.foodCategoryId} style={{width:'100%', backgroundColor: '#b6e052', borderRadius: 10, paddingLeft: 4, paddingRight: 2}}>
                      <li style={{listStyleType: 'none',justifyContent:'space-between',display:'flex', marginTop:'10px'}}>
                          <div style={{paddingTop: 7}}>Nombre: {this.getFoodCategoryNameOfObjective(objective)} Objetivo:{objective.objectiveCalories}</div>
                          <div><img src={Delete} alt="tacho" style={{width: 18, margin: 10, borderRadius: 2, cursor: 'pointer'}} onClick={() => this.props.onRemoveObjectiveFromCurrentGoal(objective)}/></div>
                      </li>
                    </div>
                  )}
                </ul>
              </div>
              <div className="w-100"></div>
              <div className="col" style={{ textAlign: 'center',marginTop:'15px' }}>
                <DatePicker
                  name='fecha consumo'
                  selected={this.state.dateStart}
                  onChange={(date) => this.handleChangeDateStart(date)}
                  dateFormat="MM-yyyy"
                  placeholderText='Fecha de consumo'
                  showMonthYearPicker
                />
              </div>
            </div>
            {this.props.history.location.state 
            ? <button onClick={this.handleSubmitUpdate} className='btn btn--primary btn--s' style={{ fontSize: '14px', boxShadow: '0px 4px 4px grey', margin: '10px', backgroundColor:'#E55812', color: 'white' }}>Editar Meta</button>
            : <button onClick={this.handleSubmitCreate} className='btn btn--primary btn--s' style={{ fontSize: '14px', boxShadow: '0px 4px 4px grey', margin: '10px', backgroundColor:'#E55812', color: 'white' }}>Crear Meta</button>
            }
            <p>{this.state.successMessage}</p>
            {this.props.isPending
              ? <div className="spinner-border" role="status">
                  <span className="sr-only"></span>
                </div>
              : null}
            <button type="button" onClick={() => {this.props.history.push("/profile"); this.props.onResetCurrentGoal()}} style={{marginTop: '10%'}}>Volver al Perfil</button>
          </div>
        </div>
        <Footer />
      </div >
    );
  }
}

export default GoalsMainPage
