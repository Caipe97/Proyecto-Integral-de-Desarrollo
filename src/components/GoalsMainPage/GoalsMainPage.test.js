import React from 'react';
import { shallow } from 'enzyme'; 
import GoalsMainPage from './GoalsMainPage';

describe('GoalsMainPage tests', () => {
    const historyMock = { push: jest.fn(),location: {state:{goal:{
        goalId: 1,
        name: 'myGoal',
        dateStart: '2021-07',
        totalCalories: 5000,
        objectives: 
        [
            {
                objectiveCalories: 3000,
                foodCategoryId: 1
            },
            {
                objectiveCalories: 2000,
                foodCategoryId: 2
            },
        ]
    }}}};
    const historyMock2 = { push: jest.fn(),location: {state:{goal: {}}}};
    //const historyMock3 = { push: jest.fn(),location: false}
    const mockProps = {
        currentGoal: {objectives: []},
        onLogout: jest.fn(),
        onAddObjectiveToCurrentGoal:jest.fn(),
        onAddGoal: jest.fn(),
        onGetAllFoods: jest.fn(),
        onLogout: jest.fn(),
        onResetCurrentGoal: jest.fn(),
        onChangeCurrentGoalNameTotalCaloriesAndDateStart: jest.fn(),
        onGetFoodCategories: jest.fn(),
        onRemoveObjectiveFromCurrentGoal:jest.fn(),
        onUpdateCurrentGoal: jest.fn(),
    };
    const mockProps2 = {
        currentGoal: {objectives: []},
        onLogout: jest.fn(),
        onAddObjectiveToCurrentGoal:jest.fn(),
        onAddGoal: jest.fn(),
        onGetAllFoods: jest.fn(),
        onLogout: jest.fn(),
        onResetCurrentGoal: jest.fn(),
        onChangeCurrentGoalNameTotalCaloriesAndDateStart: jest.fn(),
        onGetFoodCategories: jest.fn(),
        onRemoveObjectiveFromCurrentGoal:jest.fn(),
        onUpdateCurrentGoal: jest.fn(),
    };
    let wrapper,wrapper2,wrapper3;
    

    beforeEach(() => {
        wrapper = shallow(<GoalsMainPage {...mockProps} history={historyMock}/>);

        wrapper2 = shallow(<GoalsMainPage {...mockProps2} history={historyMock2}/>);
        //wrapper3 = shallow(<GoalsMainPage {...mockProps2} history={historyMock3}/>);
    })

    it('expect to render GoalsMainPage component to render', async() => {
        expect(wrapper).toMatchSnapshot();//if
        await wrapper.instance().componentDidMount();
    })
    it('expect to render GoalsMainPage component to render', async() => {
        expect(wrapper2).toMatchSnapshot();//if
        await wrapper2.instance().componentDidMount();
    })
    // it('expect to render GoalsMainPage component to render when you dont have goals state false for you', async() => {
    //     expect(wrapper3).toMatchSnapshot();//if
    //     await wrapper3.instance().componentDidMount();
    // })
    // it('expect to logout and go to login when pressing the button', () => {
    //     wrapper.find('[name="link"]').at(0).simulate('click');
    //     expect(wrapper.instance().props.onLogout).toHaveBeenCalledTimes(1);
    //     expect(wrapper.instance().props.history.push).toHaveBeenCalledTimes(1);
    //     expect(historyMock.push.mock.calls[0]).toEqual(['/']);
    // })
    it('expect to change  onclick buttons', async () => {
       
        await wrapper.find('[type="button"]').at(0).simulate('click');
       // await wrapper.find('[type="button"]').at(2).simulate('click');
       expect(wrapper.find('div')).toBeDefined();
        expect(wrapper.find('div.p').contains('')).toBeDefined();
      
    })
    it('handleChange should update the state correctly when writing on the inputs', () => {
        wrapper.find('[name="name"]').at(0).simulate('change', { target: { name: 'name', value: 'LOGROS ENERO' } });
        wrapper.find('[name="totalCalories"]').at(0).simulate('change', { target: { name: 'totalCalories', value: '123456' } });
       // expect(wrapper.instance().state).toEqual({email: 'julianlivrone@gmail.com', password: 'asd', errorMessage: ''});
    })
    it('expect to change  onclick buttons',  () => {
       wrapper.find('img').at(0).simulate('click');
       // await wrapper.find('[type="button"]').at(2).simulate('click');
      
      
    })
    // it('expect to change  onclick buttons',  () => {
    //     wrapper.find('[className="btn btn--primary btn--s"]').at(1).simulate('click');
    //     // await wrapper.find('[type="button"]').at(2).simulate('click');
       
       
    //  })
   
})
