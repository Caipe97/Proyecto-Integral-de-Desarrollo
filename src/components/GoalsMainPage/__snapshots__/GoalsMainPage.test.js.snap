// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoalsMainPage tests expect to render GoalsMainPage component to render 1`] = `
<div
  className="contenedorR"
>
  <Header
    currentGoal={
      Object {
        "objectives": Array [],
      }
    }
    history={
      Object {
        "location": Object {
          "state": Object {
            "goal": Object {
              "dateStart": "2021-07",
              "goalId": 1,
              "name": "myGoal",
              "objectives": Array [
                Object {
                  "foodCategoryId": 1,
                  "objectiveCalories": 3000,
                },
                Object {
                  "foodCategoryId": 2,
                  "objectiveCalories": 2000,
                },
              ],
              "totalCalories": 5000,
            },
          },
        },
        "push": [MockFunction],
      }
    }
    onAddGoal={[MockFunction]}
    onAddObjectiveToCurrentGoal={[MockFunction]}
    onChangeCurrentGoalNameTotalCaloriesAndDateStart={[MockFunction]}
    onGetAllFoods={[MockFunction]}
    onGetFoodCategories={[MockFunction]}
    onLogout={[MockFunction]}
    onRemoveObjectiveFromCurrentGoal={[MockFunction]}
    onResetCurrentGoal={[MockFunction]}
    onUpdateCurrentGoal={[MockFunction]}
  />
  <div
    className="contenidoR"
    style={
      Object {
        "backgroundColor": "#B6E052",
      }
    }
  >
    <div
      className="col2R"
    >
      <FoodCategoriesSearchBar
        currentGoal={
          Object {
            "objectives": Array [],
          }
        }
        history={
          Object {
            "location": Object {
              "state": Object {
                "goal": Object {
                  "dateStart": "2021-07",
                  "goalId": 1,
                  "name": "myGoal",
                  "objectives": Array [
                    Object {
                      "foodCategoryId": 1,
                      "objectiveCalories": 3000,
                    },
                    Object {
                      "foodCategoryId": 2,
                      "objectiveCalories": 2000,
                    },
                  ],
                  "totalCalories": 5000,
                },
              },
            },
            "push": [MockFunction],
          }
        }
        onAddGoal={[MockFunction]}
        onAddObjectiveToCurrentGoal={[MockFunction]}
        onChangeCurrentGoalNameTotalCaloriesAndDateStart={[MockFunction]}
        onGetAllFoods={[MockFunction]}
        onGetFoodCategories={[MockFunction]}
        onLogout={[MockFunction]}
        onRemoveObjectiveFromCurrentGoal={[MockFunction]}
        onResetCurrentGoal={[MockFunction]}
        onUpdateCurrentGoal={[MockFunction]}
      />
    </div>
  </div>
  <div
    className="sidebarR"
  >
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "column",
          "justifyContent": "center",
        }
      }
    >
      <img
        alt="goal"
        src="goal.png"
        style={
          Object {
            "width": 100,
          }
        }
      />
      <div
        style={
          Object {
            "marginTop": 15,
          }
        }
      >
        <WithStyles(ForwardRef(TextField))
          label="Nombre de la meta"
          name="name"
          onChange={[Function]}
          required={true}
          type="name"
          value="myGoal"
        />
        <WithStyles(ForwardRef(TextField))
          label="Calorías de la meta"
          name="totalCalories"
          onChange={[Function]}
          required={true}
          type="totalCalories"
          value={5000}
        />
      </div>
      <div
        className="row"
      >
        <div
          className="w-100"
        />
        <div
          className="col"
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          <ul
            style={
              Object {
                "marginBlock": "0em",
                "paddingInlineStart": "1%",
              }
            }
          />
        </div>
        <div
          className="w-100"
        />
        <div
          className="col"
          style={
            Object {
              "marginTop": "15px",
              "textAlign": "center",
            }
          }
        >
          <r
            allowSameDay={false}
            customTimeInput={null}
            dateFormat="MM-yyyy"
            dateFormatCalendar="LLLL yyyy"
            disabled={false}
            disabledKeyboardNavigation={false}
            dropdownMode="scroll"
            enableTabLoop={true}
            excludeScrollbar={true}
            focusSelectedMonth={false}
            monthsShown={1}
            name="fecha consumo"
            nextMonthButtonLabel="Next Month"
            nextYearButtonLabel="Next Year"
            onBlur={[Function]}
            onCalendarClose={[Function]}
            onCalendarOpen={[Function]}
            onChange={[Function]}
            onClickOutside={[Function]}
            onFocus={[Function]}
            onInputClick={[Function]}
            onInputError={[Function]}
            onKeyDown={[Function]}
            onMonthChange={[Function]}
            onSelect={[Function]}
            onYearChange={[Function]}
            placeholderText="Fecha de consumo"
            preventOpenOnFocus={false}
            previousMonthButtonLabel="Previous Month"
            previousYearButtonLabel="Previous Year"
            readOnly={false}
            renderDayContents={[Function]}
            selected={1625108400000}
            shouldCloseOnSelect={true}
            showFourColumnMonthYearPicker={false}
            showFullMonthYearPicker={false}
            showMonthYearPicker={true}
            showPopperArrow={true}
            showPreviousMonths={false}
            showQuarterYearPicker={false}
            showTimeInput={false}
            showTimeSelect={false}
            showTwoColumnMonthYearPicker={false}
            showYearPicker={false}
            strictParsing={false}
            timeCaption="Time"
            timeInputLabel="Time"
            timeIntervals={30}
            withPortal={false}
            yearItemNumber={12}
          />
        </div>
      </div>
      <button
        className="btn btn--primary btn--s"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#E55812",
            "boxShadow": "0px 4px 4px grey",
            "color": "white",
            "fontSize": "14px",
            "margin": "10px",
          }
        }
      >
        Editar Meta
      </button>
      <p />
      <button
        onClick={[Function]}
        style={
          Object {
            "marginTop": "10%",
          }
        }
        type="button"
      >
        Volver al Perfil
      </button>
    </div>
  </div>
  <Footer />
</div>
`;

exports[`GoalsMainPage tests expect to render GoalsMainPage component to render 2`] = `
<div
  className="contenedorR"
>
  <Header
    currentGoal={
      Object {
        "objectives": Array [],
      }
    }
    history={
      Object {
        "location": Object {
          "state": Object {
            "goal": Object {},
          },
        },
        "push": [MockFunction],
      }
    }
    onAddGoal={[MockFunction]}
    onAddObjectiveToCurrentGoal={[MockFunction]}
    onChangeCurrentGoalNameTotalCaloriesAndDateStart={[MockFunction]}
    onGetAllFoods={[MockFunction]}
    onGetFoodCategories={[MockFunction]}
    onLogout={[MockFunction]}
    onRemoveObjectiveFromCurrentGoal={[MockFunction]}
    onResetCurrentGoal={[MockFunction]}
    onUpdateCurrentGoal={[MockFunction]}
  />
  <div
    className="contenidoR"
    style={
      Object {
        "backgroundColor": "#B6E052",
      }
    }
  >
    <div
      className="col2R"
    >
      <FoodCategoriesSearchBar
        currentGoal={
          Object {
            "objectives": Array [],
          }
        }
        history={
          Object {
            "location": Object {
              "state": Object {
                "goal": Object {},
              },
            },
            "push": [MockFunction],
          }
        }
        onAddGoal={[MockFunction]}
        onAddObjectiveToCurrentGoal={[MockFunction]}
        onChangeCurrentGoalNameTotalCaloriesAndDateStart={[MockFunction]}
        onGetAllFoods={[MockFunction]}
        onGetFoodCategories={[MockFunction]}
        onLogout={[MockFunction]}
        onRemoveObjectiveFromCurrentGoal={[MockFunction]}
        onResetCurrentGoal={[MockFunction]}
        onUpdateCurrentGoal={[MockFunction]}
      />
    </div>
  </div>
  <div
    className="sidebarR"
  >
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "column",
          "justifyContent": "center",
        }
      }
    >
      <img
        alt="goal"
        src="goal.png"
        style={
          Object {
            "width": 100,
          }
        }
      />
      <div
        style={
          Object {
            "marginTop": 15,
          }
        }
      >
        <WithStyles(ForwardRef(TextField))
          label="Nombre de la meta"
          name="name"
          onChange={[Function]}
          required={true}
          type="name"
        />
        <WithStyles(ForwardRef(TextField))
          label="Calorías de la meta"
          name="totalCalories"
          onChange={[Function]}
          required={true}
          type="totalCalories"
        />
      </div>
      <div
        className="row"
      >
        <div
          className="w-100"
        />
        <div
          className="col"
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          <ul
            style={
              Object {
                "marginBlock": "0em",
                "paddingInlineStart": "1%",
              }
            }
          />
        </div>
        <div
          className="w-100"
        />
        <div
          className="col"
          style={
            Object {
              "marginTop": "15px",
              "textAlign": "center",
            }
          }
        >
          <r
            allowSameDay={false}
            customTimeInput={null}
            dateFormat="MM-yyyy"
            dateFormatCalendar="LLLL yyyy"
            disabled={false}
            disabledKeyboardNavigation={false}
            dropdownMode="scroll"
            enableTabLoop={true}
            excludeScrollbar={true}
            focusSelectedMonth={false}
            monthsShown={1}
            name="fecha consumo"
            nextMonthButtonLabel="Next Month"
            nextYearButtonLabel="Next Year"
            onBlur={[Function]}
            onCalendarClose={[Function]}
            onCalendarOpen={[Function]}
            onChange={[Function]}
            onClickOutside={[Function]}
            onFocus={[Function]}
            onInputClick={[Function]}
            onInputError={[Function]}
            onKeyDown={[Function]}
            onMonthChange={[Function]}
            onSelect={[Function]}
            onYearChange={[Function]}
            placeholderText="Fecha de consumo"
            preventOpenOnFocus={false}
            previousMonthButtonLabel="Previous Month"
            previousYearButtonLabel="Previous Year"
            readOnly={false}
            renderDayContents={[Function]}
            selected={NaN}
            shouldCloseOnSelect={true}
            showFourColumnMonthYearPicker={false}
            showFullMonthYearPicker={false}
            showMonthYearPicker={true}
            showPopperArrow={true}
            showPreviousMonths={false}
            showQuarterYearPicker={false}
            showTimeInput={false}
            showTimeSelect={false}
            showTwoColumnMonthYearPicker={false}
            showYearPicker={false}
            strictParsing={false}
            timeCaption="Time"
            timeInputLabel="Time"
            timeIntervals={30}
            withPortal={false}
            yearItemNumber={12}
          />
        </div>
      </div>
      <button
        className="btn btn--primary btn--s"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#E55812",
            "boxShadow": "0px 4px 4px grey",
            "color": "white",
            "fontSize": "14px",
            "margin": "10px",
          }
        }
      >
        Editar Meta
      </button>
      <p />
      <button
        onClick={[Function]}
        style={
          Object {
            "marginTop": "10%",
          }
        }
        type="button"
      >
        Volver al Perfil
      </button>
    </div>
  </div>
  <Footer />
</div>
`;
